python/cudf/cudf/core/column/column.py

```diff
@@ -1652,28 +1652,26 @@ def as_column(arbitrary, nan_as_null=None, dtype=None, length=None):
                     sr = pd.Series(arbitrary, dtype="str")
                     data = as_column(sr, nan_as_null=nan_as_null)
                 else:
-                    try:
-                        tmp_data = np.asarray(
+                    if dtype is None and pd.api.types.infer_dtype(
+                        arbitrary
+                    ) in ("mixed", "mixed-integer"):
+                        data = np.asarray(arbitrary, dtype="object")
+                        try:
+                            data = as_column(
+                                data, dtype=dtype, nan_as_null=nan_as_null
+                            )
+                        except BaseException:
+                            raise TypeError(
+                                "CUDF doesn't support incompatible mixed types"
+                            )
```

Comment:
I think it would be better to let it raise the PyArrow or Numpy exception here as they'll likely be more indicative of the problem going on than this generic message.

Alternatively we could take the error message from their except and raise our own `TypeError` here but that feels fragile.

Reply:
I like option 1